src/constantProductCalculator.ts

```diff
@@ -101,4 +148,37 @@ export class ConstantProductCalculator implements SwapCalculator {
       "Amount of minted liquidity tokens must be greater then 0.",
     );
   }
+
+  getZapParams(zapAmount: bigint, input: ZapInput): ZapParams {
```

Comment:
1. In other places we pass a specific asset we want to perform an action on. `getZapParams(asset: Asset, amount: bigint)`
2. The algorithm should also be exposed as a pure function.

```ts
export function getZapParams(liqA: bigint, liqB: bigint, amount: bigint, feeBps: bigint): ZapParams
```
```